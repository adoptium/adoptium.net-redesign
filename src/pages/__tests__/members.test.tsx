import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it, vi } from "vitest"
import { axe } from "vitest-axe"
import Members, { Head } from "../members"

vi.mock("../../components/Logos", () => {
  return {
    __esModule: true,
    default: () => <div />,
    LogoType: {},
  }
})

describe("Members Page", () => {
  it("renders correctly", () => {
    const { container } = render(<Members />)
    expect(container).toMatchSnapshot()
  })

  it("head renders correctly", () => {
    const { container } = render(<Head />)
    // eslint-disable-next-line
    const title = container.querySelector("title")
    expect(title?.textContent).toEqual("Our Members | Adoptium")
  })

  it("has no accessibility violations", async () => {
    const { container } = render(<Members />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
