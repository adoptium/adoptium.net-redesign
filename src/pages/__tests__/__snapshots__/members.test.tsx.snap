// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Members Page > renders correctly 1`] = `
<div>
  <div
    class="slice--banner"
  />
  <div
    class="slice--navbar"
  />
  <main>
    <div
      class="relative pt-40 pb-16 md:pb-36 md:pt-60 overflow-hidden"
    >
      <div
        class="w-full relative"
      >
        <div
          class="absolute z-[-1] w-[2396px] h-[1340px] left-[23px] md:left-[112px] top-[21px] md:top-[171px] bg-[#410170] shadow-[0_0_400px_rgba(65,1,112,1)] rounded-full blur-[400px]"
        />
        <div
          class="absolute z-[-1] w-[1487px] h-[893px] left-[120px] md:left-[676px] top-[120px] md:top-[395px] bg-[#B62987] shadow-[0_0_400px_rgba(182,41,135,1)] rounded-full blur-[400px]"
        />
        <div
          class="absolute z-[-1] w-[688px] h-[446px] left-[400px] md:left-[1131px] top-[618px] bg-[#FE8492] shadow-[0_0_400px_rgba(254,132,146,1)] rounded-full blur-[400px]"
        />
      </div>
      <div
        class="mx-auto max-w-[1048px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
      >
        <div
          class="self-stretch flex-col justify-center items-center gap-6 flex"
        >
          <div
            class="self-stretch  flex-col justify-center items-center gap-4 flex"
          >
            <div
              class="justify-start items-center gap-3 inline-flex"
            >
              <svg
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#FF1464"
                  height="8"
                  rx="2"
                  width="8"
                  x="3"
                  y="3"
                />
                <rect
                  height="11"
                  rx="3.5"
                  stroke="#FF1464"
                  stroke-opacity="0.25"
                  stroke-width="3"
                  width="11"
                  x="1.5"
                  y="1.5"
                />
              </svg>
              <div
                class="text-rose-600 text-base font-bold leading-normal"
              >
                Our Members
              </div>
            </div>
            <div
              class="self-stretch text-center text-white   text-[56px] lg:text-[80px] leading-[114.286%] md:leading-[120%] font-semibold"
            >
              Who we work with
            </div>
          </div>
          <div
            class="self-stretch text-center text-lightgrey text-[20px] font-normal leading-[140%] undefined"
          >
            Eclipse Temurin offers high-performance, cross-platform, open-source Java runtime binaries that are enterprise-ready and Java SE TCK-tested for general use in the Java ecosystem.
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex gap-4 md:gap-8 items-center justify-center absolute w-full top-[571px]"
    >
      <a
        href="/en/join"
      >
        <button
          class="rounded-2xl bg-transparent gradient-border border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[179px] h-[48px]  transition-all duration-500 ease-in-out mt-8"
        >
          Become a Member
        </button>
      </a>
      <button
        class="rounded-2xl bg-transparent gradient-border border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[179px] h-[48px]  transition-all duration-500 ease-in-out mt-8"
      >
        Our Members
      </button>
    </div>
    <div
      class="overflow-auto px-3 py-5 w-full md:w-auto"
    >
      <h3
        class="flex space-x-10 whitespace-nowrap  justify-center py-2"
      >
        <a
          href="#strategic-sec"
        >
          <span
            class="text-[16px] py-2 border-primary w-full text-base font-normal leading-6
                outline-none hover:text-white hover:border-b
                hover:border-[#ff1464] text-[#8a809e] border-b
                border-transparent cursor-pointer transition-all duration-200 ease-in-out"
          >
            Strategic Members
          </span>
        </a>
        <a
          href="#enterprise-sec"
        >
          <span
            class="text-[16px] py-3 border-primary w-full text-base font-normal leading-6
                outline-none hover:text-white hover:border-b
                hover:border-[#ff1464] text-[#8a809e] border-b
                border-transparent cursor-pointer transition-all duration-200 ease-in-out"
          >
            Enterprise Members
          </span>
        </a>
        <a
          href="#participant-sec"
        >
          <span
            class="text-[16px] py-2 border-primary w-full text-base font-normal leading-6
                outline-none hover:text-white hover:border-b
                hover:border-[#ff1464] text-[#8a809e] border-b
                border-transparent cursor-pointer transition-all duration-200 ease-in-out"
          >
            Participant Members
          </span>
        </a>
      </h3>
    </div>
    <div />
    <div />
    <div />
    <div
      class="flex justify-center flex-wrap items-center gap-5 md:gap-14 m-5"
    >
      <p
        class="text-[20px] leading-[28px] text-white my-0 text-center"
      >
        Are you interested in becoming a member?
      </p>
      <a
        href="/en/join"
      >
        <button
          class="rounded-2xl bg-transparent gradient-border border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[179px] h-[48px]  transition-all duration-500 ease-in-out "
        >
          Become a Member
        </button>
      </a>
    </div>
    <section
      class="relative  xl:px-0 px-6 md:py-32 py-16 undefined"
    >
      <img
        alt="gradient overlay"
        class="hidden md:block absolute bg-center top-0 left-0  right-0 w-full h-full z-[-10]"
        src="/images/gradient-overlay.png"
      />
      <img
        alt="gradient overlay mobile"
        class="block md:hidden absolute bg-center top-0 left-0  right-0 w-full h-full z-[-10]"
        src="/images/gradient-overlay-mobile.png"
      />
      <div
        class="max-w-[832px] w-full mx-auto md:py-0 py-8 flex justify-center items-center flex-col text-center md:bg-transparent bg-tones-white rounded-3xl md:backdrop-blur-0 backdrop-blur-12 "
      >
        <img
          aria-label="message icon"
          class="!mb-0"
          src="/images/icons/message.svg"
        />
        <p
          class="text-white text-4xl md:text-5xl leading-[122.222%] md:leading-[116.667%] font-normal mt-8 mb-6"
        >
          Speak to our team today
        </p>
        <span
          class=" text-lightgrey tab-button-text"
        >
          Ready to get involved? Fill out our membership form and take the first step in shaping the future of Adoptium.
        </span>
        <a
          href="/en/join"
        >
          <button
            class="rounded-2xl bg-transparent gradient-border border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[179px] h-[48px]  transition-all duration-500 ease-in-out  hidden md:block mt-8"
          >
            Contact Us
          </button>
        </a>
      </div>
    </section>
  </main>
  <div
    class="bg-blue border-[#3E3355] border-b py-10 px-6"
  >
    <div
      class="flex flex-col lg:flex-row max-w-[1048px] mx-auto justify-center lg:justify-between gap-5 items-center "
    >
      <p
        class="text-3xl md:text-4xl leading-[38px] md:leading-[48px] text-white my-0 text-center"
      >
        Thank you to our 
        <span
          class="text-pink pr-1"
        >
          300+
        </span>
        contributors
      </p>
      <div
        class="flex items-center justify-center p-4 space-x-4 w-[325px] rounded-3xl border border-white/50"
      />
    </div>
  </div>
  <div
    class="slice--footer"
  />
</div>
`;
